tests/python/gpu/test_gluon_gpu.py

```diff
@@ -233,6 +236,47 @@ def test_symbol_block_fp16():
             break
     assert np.dtype(net_fp16.params[param_name].dtype) == np.dtype(np.float16)
 
+
+@with_seed()
+def test_large_models():
+    ctx = default_context()
+    # Create model
+    net = gluon.nn.HybridSequential()
+
+    largest_num_features = 256
+    with net.name_scope():
+        net.add(nn.Conv2D(128, 3))
+        net.add(nn.LeakyReLU(0.1))
+        net.add(nn.Conv2D(largest_num_features, 3))
+        net.add(nn.LeakyReLU(0.1))
+        net.add(nn.Conv2D(1, 3))
+
+    net.hybridize()
+    net.initialize(mx.init.Normal(sigma=0.01), ctx=ctx)
+    mx.nd.waitall()
```

Comment:
since the in channels are not defined and the network didn't go through a forward, deferred initialization won't happen and no actual allocation is happening here.

Reply:
I borrowed the approach from the mentioned issue, what about the lines:

https://github.com/apache/incubator-mxnet/blob/94614a545d0191cbbea95d944e470004eb5e9839/tests/python/gpu/test_gluon_gpu.py#L276-L277

Reply:
the first time it runs through this line of code, it will trigger actual allocation. I'm merely pointing out that the waitall is likely not needed.